source/DurationPicker.js

```diff
@@ -0,0 +1,781 @@
+			}
+		},
+
+		
+		/**
+		* @private
+		*/
+		valueChanged: function () {
+			if (this.value == null || this.value === '') {
+				this.resetPicker();
+				this.noneTextChanged();
+				this.doDurationChange({name: this.name, value: this.$.currentValue.content});
+			} else {
+				this.createValues();
+				this.updatePicker();
+				this.$.currentValue.set('content', this.formatText());
+			}
+		},
+
+		/**
+		* @private
+		*/
+		createValues: function () {
+			var tempValue;
+			if (this.value != null && this.value !== '') {
+				var timeArray = this.value.toString().split(':');
+				for (var idx = 0; idx < this.templateArray.length; idx++) {
+					var item = this.templateArray[idx];
+					switch (item) {
+						case 'h':
+						if (this.$.hourPicker || !this.$.hourPicker && timeArray.length > 2) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue > 23 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.values['hour'] = tempValue;
+						}
+						break;
+						case 'm':
+						if (this.$.minutePicker || !this.$.minutePicker && timeArray.length > 1) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue > 59 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.values['minute'] = tempValue;
+						}
+						break;
+						case 's':
+						if (this.$.secondPicker || !this.$.secondPicker && timeArray.length > 0) {
+							tempValue = Math.round(timeArray.shift());
+							if (isNaN(tempValue) || tempValue> 59 || tempValue < 0) {
+								tempValue = 0;
+							}
+							this.values['second'] = tempValue;
+						break;
+						}
+					}
+				}
+			}
+		},
+
+		/**
+		* @private
+		*/
+		updatePicker: function () {
+			if (this.$.hourPicker && this.$.hour.value != this.values['hour'])
+				this.$.hour.set('value', this.values['hour']);
+
+			if (this.$.minutePicker && this.$.minute.value != this.values['minute'])
+				this.$.minute.set('value', this.values['minute']);
+
+			if (this.$.secondPicker && this.$.second.value != this.values['second'])
+				this.$.second.set('value',this.values['second']);
+		},
+
+		/**
+		* @private
+		*/
+		formatValue: function () {
+			var item, idx, val = '';
+			for (idx = 0; idx < this.templateArray.length; idx++) {
+				item = this.templateArray[idx];
+				switch (item) {
+					case 'h':
+						val = val + ':' + this.values['hour'];
+						break;
+					case 'm':
+						val = val + ':' + this.values['minute'];
+						break;
+					case 's':
+						val = val + ':' + this.values['second'];
+						break;
+				}
+			}
+			return val.substring(1);
+		},
+
+		/**
+		* @private
+		*/
+		formatText: function () {
+			var item, idx, text = '';
+			if (this.values) {
+				for (idx = 0; idx < this.templateArray.length; idx++) {
+					item = this.templateArray[idx];
+					switch (item) {
+						case 'h':
+							text = text + ' ' + this.values['hour'] + ' ' + (this.values['hour'] == 1 ? moon.$L('Hour') : moon.$L('Hours'));
```

Comment:
Hi @anishlg, instead of using `moon.$L('Hour')` and `moon.$L('Hours')` here, can we instead utilize our `hourText` property? Similarly, we would want to do the same thing for minutes and seconds.